### ISSUE TITLE
Flutter Web screen recording tools are producing subpar results due to flutter web's rendering

### ISSUE THREAD

**Issue by idotenenbaum** [flutter/flutter#84690]
TL;DR -- our flutter web Hotjar and Smartlook screen recording are producing subpar results due to flutter web's rendering. 

Context:

* We're running our frontend in Flutter Web.

* We'd like to install a screen recording tool that would record the sessions of our users.

* We tried several tools (Hotjar being one of them), but all face a very similar problem: there's something about the way Flutter renders the webpage that makes it difficult for the screen-recording tool to properly capture the elements on the page.

* Example of the result is can be viewed in the attached images; the image isn't rendering. This is a minor example, but we have many more.

Actual webpageview:

<img width="1440" alt="actual-webpage-view" src="https://user-images.githubusercontent.com/73842765/122188888-186c4900-ce99-11eb-99e3-47db521b5a0a.png">

Screen recording result:

<img width="1440" alt="hotjar-screen-recording-result" src="https://user-images.githubusercontent.com/73842765/122188898-1a360c80-ce99-11eb-9b72-dfefa2be2329.png">

* We've tried using the HTML web renderer (https://flutter.dev/docs/development/tools/web-renderers) -- improvement was unfortunately very small (above picture is with the renderer).

This is an important enterprise feature that is blocking our product work. Would appreciate your help on this.

**Comment by yjbanov** [comment]
I don't know how tools like that work. I suspect they may be relying on semantic information in the UI. Do you get better results when you enable semantics? You can do that as follows:

```dart
main() {
  runApp(...);
  RendererBinding.instance!.setSemanticsEnabled(true);
}
```

**Comment by yjbanov** [comment]
The best integration with Flutter might be for Hotjar and Smartlook to provide Flutter packages that probe into the widget hierarchy (using something like `WidgetController` but without a dependency on `package:test` so it can be deployed to production) rather than try to reverse-engineer and replay the HTML we generate.

**Comment by darshankawar** [comment]
@idotenenbaum 
This seems to sound more of not in Flutter's control, but the way `Hotjar` and `Smartlook` currently handles the framework.
Please look at above comments and see if either of them helps.

**Comment by tjerkw** [comment]
Smartlook does have a flutter SDK for Web and iOS: https://smartlook.github.io/docs/sdk/flutter

But if you use the Web (non-flutter) integration then the recording is a white screen. 

I don't know why this is yet.

**Comment by TahaTesser** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
